asf_search/ASFSearchOptions/validators.py

```diff
@@ -126,27 +126,27 @@ def parse_number_or_range(value: Union[list, Tuple[number, number]], h):
         raise ValueError(f'Invalid {h.__name__} or range: {exc}') from exc
 
 
-# Parse and validate a list of numbers or number ranges, using h() to validate each value: "1,2,3-5", "1.1,1.4,5.1-6.7"
-def parse_number_or_range_list(value: list, h) -> list:
-    if not isinstance(value, list):
+# Parse and validate an iterable of numbers or number ranges, using h() to validate each value: "1,2,3-5", "1.1,1.4,5.1-6.7"
+def parse_number_or_range_iterator(value: Iterable, h) -> list:
+    if not isinstance(value, Iterable):
         value = [value]
     return [parse_number_or_range(x, h) for x in value]
 
 
-# Parse and validate a list of integers or integer ranges: "1,2,3-5"
-def parse_int_or_range_list(value: list) -> list:
-    return parse_number_or_range_list(value, int)
+# Parse and validate an iterable of integers or integer ranges: "1,2,3-5"
+def parse_int_or_range_iterator(value: Iterable) -> list:
+    return parse_number_or_range_iterator(value, int)
 
 
-# Parse and validate a list of float or float ranges: "1.0,2.0,3.0-5.0"
-def parse_float_or_range_list(value: list) -> list:
-    return parse_number_or_range_list(value, parse_float)
+# Parse and validate an iterable of float or float ranges: "1.0,2.0,3.0-5.0"
+def parse_float_or_range_iterator(value: Iterable) -> list:
```

Comment:
All of these have `list` instead of `List`. Does python 3.8 support it if it's in the return? It probably needs to be an all or nothing for 3.8 to work I'm guessing